### ISSUE TITLE
License Finder crashes on nested directories it considers "candidate files"

### ISSUE THREAD

**Issue by pmeskers** [pivotal/LicenseFinder#143]
Hey there,

We just pulled in the most recent release (2.0.2) and had it crash on execution. It appears that license finder is finding a list of candidate files, and then iterating over this list to see if each file is a directory or a file. However, when there is a directory inside one of the candidate directories, license finder will *not* realize it is a directory, and instead attempt to check if it is a license (this crashes the app, EISDIR).

We've reproduced this with a failing spec, but don't have the bandwidth to fix it at this point in time. You can see this [here](https://github.com/pmeskers/LicenseFinder/tree/broken-nested-behavior). 

It should be noted this is a pretty serious issue because any projects which have license_finder as a dependency will actually fail (it fails when run against itself for the reason listed above!).

The full backtrace looks like this:

```
$ license_finder
.........................................................................................................................................................................................................................................................................................................................................
/Users/pivotal/workspace/LicenseFinder/lib/license_finder/packages/possible_license_file.rb:16:in `binread': Is a directory @ io_fread - /Users/pivotal/workspace/LicenseFinder/lib/license_finder/cli (Errno::EISDIR)
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/packages/possible_license_file.rb:16:in `binread'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/packages/possible_license_file.rb:16:in `text'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/packages/possible_license_file.rb:12:in `license'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/packages/license_files.rb:19:in `block in find'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/packages/license_files.rb:19:in `reject'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/packages/license_files.rb:19:in `find'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/packages/license_files.rb:9:in `find'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/package.rb:113:in `license_files'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/package.rb:99:in `licensing'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/package.rb:93:in `activations'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/package.rb:89:in `licenses'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/decision_applier.rb:42:in `with_approval'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/decision_applier.rb:24:in `block in packages'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/decision_applier.rb:24:in `map'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/decision_applier.rb:24:in `packages'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/decision_applier.rb:13:in `acknowledged'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/decision_applier.rb:9:in `unapproved'
	from /Users/pivotal/workspace/LicenseFinder/lib/license_finder/cli/main.rb:26:in `action_items'
	from /Users/pivotal/.rvm/gems/ruby-2.2.1/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
	from /Users/pivotal/.rvm/gems/ruby-2.2.1/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/pivotal/.rvm/gems/ruby-2.2.1/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
	from /Users/pivotal/.rvm/gems/ruby-2.2.1/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
	from /Users/pivotal/workspace/LicenseFinder/bin/license_finder:5:in `<top (required)>'
	from /Users/pivotal/.rvm/gems/ruby-2.2.1/bin/license_finder:23:in `load'
	from /Users/pivotal/.rvm/gems/ruby-2.2.1/bin/license_finder:23:in `<main>'
	from /Users/pivotal/.rvm/gems/ruby-2.2.1/bin/ruby_executable_hooks:15:in `eval'
	from /Users/pivotal/.rvm/gems/ruby-2.2.1/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by flavorjones** [comment]
Will take a look.

**Comment by flavorjones** [comment]
Confirmed, thanks so much for the bug report!

**Comment by flavorjones** [comment]
For posterity, this was fixed in v2.0.3.
